### PR TITLE
feat(ingest/powerbi): link cross workspace dataset into assets

### PR DESCRIPTION
**Pull request by sid-acryl**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 1]
```diff
@@ -447,6 +473,16 @@ def fill_dashboards() -> None:
                 dashboard.tiles = self._get_resolver().get_tiles(
                     workspace, dashboard=dashboard
                 )
+                # set the dataset for tiles
+                for tile in dashboard.tiles:
+                    if tile.dataset_id:
+                        tile.dataset = self.dataset_registry.get(tile.dataset_id)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:479]: is this dataset registry global across all workspaces?

also, do both workspaces need to be ingested for cross-workspace lineage to work? I think we might need some docs updates if so

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_classes.py [hunk 2]
```diff
@@ -218,6 +218,7 @@ class Report:
     webUrl: Optional[str]
     embedUrl: str
     description: str
+    dataset_id: Optional[str]
     dataset: Optional["PowerBIDataset"]
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_classes.py:222]: how do dataset_id and dataset interact? is one filled initially, and then the other populated later?

would be good to add some comments to make this more clear

**Review by hsheth2 — CHANGES_REQUESTED** [review]
had a couple questions / comments, but overall seems reasonable

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py (hunk 1 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:479]: Yes both need to be ingested. You can check the info message at powerbi_api.py:176

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:479]: Ah I see - we should make this message way more clear then

it should say that this impacts lineage for the report/tile. the title should be like "missing lineage for report", and the context should say that it's like a cross-workspace reference that failed to be resolved

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py [hunk 3]
```diff
@@ -47,6 +57,10 @@ def __init__(self, config: PowerBiDashboardSourceConfig) -> None:
             tenant_id=self.__config.tenant_id,
         )
 
+        self.reporter = reporter
+
+        self.dataset_registry = {}
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:62]: needs a type annotation + add a comment explaining what the key is

**Review by hsheth2 — CHANGES_REQUESTED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py (hunk 3 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py:62]: The detail is available from line number 35 to 38

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
@sid-acryl there's git conflicts on this?

**Review by hsheth2 — APPROVED** [review]

**Comment by sid-acryl** [comment]
> @sid-acryl there's git conflicts on this?

resolved

**Pull request merged**